EIPS/eip-7834.md

```diff
@@ -0,0 +1,92 @@
+---
+eip: 7834
+title: Separate Metadata Section for EOF
+description: Introduce a new separate metadata section to the EOF
+author: Kaan Uzdogan (@kuzdogan), Marco Castignoli (@marcocastignoli), Manuel Wedler (@manuelwedler)
+discussions-to: https://ethereum-magicians.org/t/eip-7834-separate-metadata-section-for-eof/22138
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-12-06
+requires: 3540
+---
+
+## Abstract
+
+Introduce a new separate metadata section to the Ethereum Object Format (EOF) that is unreachable by the code, and any changes to which does not affect the code.
+
+## Motivation
+
+It is desirable to include metadata in contract's bytecode for various reasons. For instance, both the Solidity and Vyper compilers by default include the language and compiler version used to compile. Solidity additionally includes the IPFS or the Swarm hash of the Solidity contract metadata.json file, and the experimental Solidity flag. The current (pre-EOF) practice is to append this CBOR encoded metadata section in the contract's runtime bytecode, followed by the 2 bytes length of the CBOR encoded bytes.
```

Comment:
fwiw, in Vyper `0.4.1` (to be released), the integrity hash (using [`sha256`](https://github.com/vyperlang/vyper/blob/12ab4919cc4618fcac4f5d24d45a0e7fdbc4a48c/vyper/utils.py#L215-L217) under to the hood) is now put into the initcode (not runtime code) (https://github.com/vyperlang/vyper/pull/4234) to preserve bytecode space of the runtime code. That's also a smart way to deal with it ;)

Reply:
Added, thanks for the info